src/pages/billing/Project.js

```diff
@@ -22,44 +22,82 @@ export default ajaxCaller(class ProjectDetail extends Component {
       editingUser: false,
       deletingUser: false,
       updating: false,
+      billingAccountName: null,
       ...StateHistory.get()
     }
   }
 
-  refresh = _.flow(
-    withErrorReporting('Error loading billing project users list'),
+  updateBillingAccount = _.flow(
+    withErrorReporting('Error updating billing account'),
     Utils.withBusyState(v => this.setState({ loading: v }))
-  )(async () => {
-    const { ajax: { Billing }, project } = this.props
-    this.setState({ addingUser: false, deletingUser: false, updating: false, editingUser: false })
-    const rawProjectUsers = await Billing.project(project.projectName).listUsers()
-    const projectUsers = _.flow(
-      _.groupBy('email'),
-      _.map(gs => ({ ..._.omit('role', gs[0]), roles: _.map('role', gs) })),
-      _.sortBy('email')
-    )(rawProjectUsers)
-    this.setState({ projectUsers })
+  )(async newAccountName => {
+    const { ajax: { GoogleBilling }, project: { projectName } } = this.props
+    const { billingAccountName } = await GoogleBilling.changeBillingAccount({ projectId: projectName, newAccountName })
+    this.setState({ billingAccountName })
   })
 
-  componentDidMount() {
-    this.refresh()
-  }
+  loadBillingInfo = withErrorReporting('Error loading current billing account',
+    async () => {
+      const { ajax: {  GoogleBilling }, project: { projectName } } = this.props
+      if (Auth.hasBillingScope()) {
+        const { billingAccountName } = await GoogleBilling.getBillingInfo(projectName)
+        this.setState({ billingAccountName })
+      }
+    })
+
+  refresh = withErrorReporting('Error loading billing project users list',
+    async () => {
+      const { ajax: { Billing }, project } = this.props
+      this.setState({ addingUser: false, deletingUser: false, updating: false, editingUser: false })
+      const rawProjectUsers = await Billing.project(project.projectName).listUsers()
+      const projectUsers = _.flow(
+        _.groupBy('email'),
+        _.map(gs => ({ ..._.omit('role', gs[0]), roles: _.map('role', gs) })),
+        _.sortBy('email')
+      )(rawProjectUsers)
+      this.setState({ projectUsers })
+    })
+
+  componentDidMount = Utils.withBusyState(
+    loading => this.setState({ loading }),
+    () => Promise.all([
+      this.refresh(),
+      this.loadBillingInfo()
+    ])
+  )
 
   render() {
-    const { project: { projectName, creationStatus }, ajax: { Billing } } = this.props
-    const { projectUsers, loading, updating, filter, addingUser, deletingUser, editingUser } = this.state
+    const { project: { projectName, creationStatus }, ajax: { Billing }, billingAccounts, authorizeAndLoadAccounts, loadingAccountAuth } = this.props
+    const { projectUsers, loading, updating, filter, addingUser, deletingUser, editingUser, billingAccountName } = this.state
     const adminCanEdit = _.filter(({ roles }) => _.includes('Owner', roles), projectUsers).length > 1
+    const billingNode = Utils.cond(
```

Comment:
Nit: `billingNode` is a little general, maybe rename it to something more specific like `selectBillingAccount`?